packages/components/src/custom-select-control-v2/legacy-component/test/index.tsx

```diff
@@ -0,0 +1,467 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import { click, press, sleep, type, waitFor } from '@ariakit/test';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import UncontrolledCustomSelect from '..';
+
+const customClass = 'amber-skies';
+
+const legacyProps = {
+	label: 'label!',
+	options: [
+		{
+			key: 'flower1',
+			name: 'violets',
+		},
+		{
+			key: 'flower2',
+			name: 'crimson clover',
+			className: customClass,
+		},
+		{
+			key: 'flower3',
+			name: 'poppy',
+		},
+		{
+			key: 'color1',
+			name: 'amber',
+			className: customClass,
+		},
+		{
+			key: 'color2',
+			name: 'aquamarine',
+			style: {
+				backgroundColor: 'rgb(127, 255, 212)',
+				rotate: '13deg',
+			},
+		},
+	],
+};
+
+const LegacyControlledCustomSelect = ( {
+	options,
+	onChange,
+	...restProps
+}: React.ComponentProps< typeof UncontrolledCustomSelect > ) => {
+	const [ value, setValue ] = useState( options[ 0 ] );
+	return (
+		<UncontrolledCustomSelect
+			{ ...restProps }
+			options={ options }
+			onChange={ ( args: any ) => {
+				onChange?.( args );
+				setValue( args.selectedItem );
+			} }
+			value={ options.find(
+				( option: any ) => option.key === value.key
+			) }
+		/>
+	);
+};
+
+describe( 'With Legacy Props', () => {
```

Comment:
The legacy component will now always receive legacy props, right? Might make sense to rename this test grouping, or to just remove it altogether.

Reply:
Good catch, fixed 👍